### PR TITLE
Fix doc syncing for `.x` removal

### PR DESCRIPTION
**Pull request by huonw**
This fixes the doc syncer now that we don't have `.x` in the directories (#48), per https://github.com/pantsbuild/pantsbuild.org/pull/48#issuecomment-1876073051

It also makes an unrelated fix, so that my test works.

Test: #57, https://github.com/pantsbuild/pantsbuild.org/actions/runs/7403672327/job/20143898054

### CODE REVIEW

File: .github/workflows/sync_docs.yml [hunk 1]
```diff
@@ -49,17 +49,15 @@ jobs:
...
         id: get-destination-dir
         run: |
-          DESTINATION_DIR="versioned_docs/version-${{ steps.get-branch-name.outputs.BRANCH_NAME }}"
+          BASE_VERSION=$(echo "${{ inputs.version }}" | sed -E 's/^([0-9]+)\.([0-9]+)\.\w+.*/\1.\2/')
+          DESTINATION_DIR="versioned_docs/version-$BASE_VERSION"
           if [ ! -d "pantsbuild.org/$DESTINATION_DIR" ]; then
             DESTINATION_DIR="docs"
           fi
-          echo "DESTINATION_DIR=$(echo "$DESTINATION_DIR")" >> "${GITHUB_OUTPUT}"
+          echo "DESTINATION_DIR=$(echo "$DESTINATION_DIR")" | tee -a "${GITHUB_OUTPUT}"
```
Comment (huonw) [inline .github/workflows/sync_docs.yml:60]: Drive-by change, for non-confidential values like this, printing the value to stdout helps with debugging, and `tee -a` is a handy way to do so without needing to repeat the `echo`s.

File: .github/workflows/sync_docs.yml [hunk 2]
```diff
@@ -49,21 +49,19 @@ jobs:
...
       - run: |
           rm -rf "pantsbuild.org/${{ steps.get-destination-dir.outputs.DESTINATION_DIR }}"
-          mkdir "pantsbuild.org/${{ steps.get-destination-dir.outputs.DESTINATION_DIR }}"
-          cp help-all.json "pantsbuild.org/${{ steps.get-destination-dir.outputs.DESTINATION_DIR }}"
+          mkdir -p "pantsbuild.org/${{ steps.get-destination-dir.outputs.DESTINATION_DIR }}/reference"
+          cp help-all.json "pantsbuild.org/${{ steps.get-destination-dir.outputs.DESTINATION_DIR }}/reference"
```
Comment (huonw) [inline .github/workflows/sync_docs.yml:64]: (This appears to be an existing issue with this action. fixed here but not directly related to #48.)

File: .github/workflows/sync_docs.yml (hunk 1 above)
Comment (thejcannon) [reply .github/workflows/sync_docs.yml:60]: I'm gonna try really hard to keep this one in my pocket.

**Review by thejcannon — APPROVED** [review]
